tests/aws/services/cloudwatch/test_cloudwatch_metrics.py

```diff
@@ -123,3 +123,37 @@ def _wait_for_lambda_metric(
         )
         assert res["MetricDataResults"][0]["Values"] == expected_return
         return res
+
+
+class TestSqsApproximateMetrics:
+    @markers.aws.validated
+    def test_sqs_approximate_metrics(self, aws_client, sqs_create_queue):
+        queue_names = []
+        for _ in range(0, 10):
+            q_name = f"my-test-queue-{short_uid()}"
+            sqs_create_queue(QueueName=q_name)
+            queue_names.append(q_name)
+
+        for queue in queue_names:
+            retry(
+                lambda: self._assert_approximate_metrics_for_queue(
+                    aws_client.cloudwatch, queue_name=queue
+                ),
+                retries=70,  # should be reported every 60 seconds on LS
+                sleep=10 if is_aws() else 1,
```

Comment:
nit: Attempting this 70 times seems odd, especially if LS only reports this after 60s (which seems to be the case). Couldn't we `sleep_before=60` before we start checking this, then have 7 retries? Or does AWS need those 70 10s intervals?

Reply:
two things:
1. we don't really know when the scheduler started (it executes the metric-reporting every 60 seconds, but starts when the SQS provider was first called), and how it long it takes to finish the reporting. 70 repeats seemed like a safe choice. but when we are lucky the metrics could already be available within a few seconds.
2. the retry is running in a loop, for each queue_name, so adding a `sleep_before` doesn't make sense either

AWS doesn't really provide any information on the interval they use, when I run tests yesterday it also seemed very random (no relation between metric report time and creation/age of the queue).